src/appfl/run_mpi.py

```diff
@@ -244,47 +251,104 @@ def run_client(
         cfg.validation = False
         test_dataloader = None
 
-    clients = [
-        eval(cfg.fed.clientname)(
-            cid,
-            weight[cid],
-            copy.deepcopy(model),
-            loss_fn,
-            DataLoader(
-                train_data[cid],
-                num_workers=cfg.num_workers,
-                batch_size=batchsize[cid],
-                shuffle=cfg.train_data_shuffle,
-                pin_memory=True,
-            ),
-            cfg,
-            outfile[cid],
-            test_dataloader,
-            **cfg.fed.args,
+    clientpergpu = math.ceil(num_clients/cfg.num_gpu)    
+    clients = []
+    for _, cid in enumerate(num_client_groups[comm_rank - 1]):
+        ## We assume to have as many GPUs as the number of MPI processes.
```

Comment:
Does this line of comment still hold?

Reply:
Yes, the assumption is still held for the benchmarking performance (probably the best performance by [this reference](https://saturncloud.io/blog/how-to-share-model-and-session-between-processes-in-inference-with-nvidia-gpu/#:~:text=GPUs%20have%20limited%20memory%20and,decreased%20performance%20and%20even%20crashes.)). However, the updated code also allows users to run a non-matching combination of the GPU and MPI process to run and utilize the GPU as much as possible.